ee/apps/omnichannel-transcript/src/OmnichannelTranscript.ts

```diff
@@ -1,16 +1,272 @@
+					// If there's no attachment and no message, what was sent? lol
+					return { _id: message._id, files: [], ts: message.ts, u: message.u, msg: message.msg };
+				}
+
+				const files = await Promise.all(
+					message.attachments.map(async (attachment) => {
+						// @ts-expect-error - messages...
+						if (attachment.type !== 'file') {
+							// ignore other types of attachments
+							return;
+						}
+						// @ts-expect-error - messages...
+						if (!this.worker.isMimeTypeValid(attachment.image_type)) {
+							// ignore invalid mime types
+							return { name: attachment.title, buffer: null };
+						}
+						const file = message.files?.find((file) => file.name === attachment.title);
+						if (!file) {
+							// ignore attachments without file
+							// This shouldn't happen, but just in case :)
+							return;
+						}
+						const uploadedFile = await Uploads.findOneById(file._id);
+						if (!uploadedFile) {
+							// ignore attachments without file
+							return { name: file.name, buffer: null };
+						}
+
+						const fileBuffer = await this.uploadService.getFileBuffer({ userId, file: uploadedFile });
+						return { name: file.name, buffer: fileBuffer };
+					}),
+				);
+
+				// When you send a file message, the things you type in the modal are not "msg", they're in "description" of the attachment
+				// So, we'll fetch the the msg, if empty, go for the first description on an attachment, if empty, empty string
+				const msg = message.msg || message.attachments.find((attachment) => attachment.description)?.description || '';
+				// Remove nulls from final array
+				return { _id: message._id, msg, u: message.u, files: files.filter(Boolean), ts: message.ts };
+			}),
+		);
+	}
+
+	async workOnPdf({ template, details }: { template: Templates; details: WorkDetailsWithSource }): Promise<void> {
+		this.log.log(`Processing transcript for room ${details.rid} by user ${details.userId} - Received from queue`);
+		if (this.maxNumberOfConcurrentJobs <= this.currentJobNumber) {
+			this.log.error(`Processing transcript for room ${details.rid} by user ${details.userId} - Too many concurrent jobs, queuing again`);
+			throw new Error('retry');
+		}
+		this.currentJobNumber++;
+
+		const room = await LivechatRooms.findOneById(details.rid);
+		if (!room) {
+			throw new Error('room-not-found');
+		}
+		const messages = await this.getMessagesFromRoom({ rid: room._id });
+
+		const data = {
+			visitor:
+				room.v && (await LivechatVisitors.findOneById(room.v._id, { projection: { _id: 1, name: 1, username: 1, visitorEmails: 1 } })),
+			agent: room.servedBy && (await Users.findOneAgentById(room.servedBy._id, { projection: { _id: 1, name: 1, username: 1 } })),
+			closedAt: room.closedAt,
+			messages: await this.getFiles(details.userId, messages),
+		};
+
+		try {
+			await this.doRender({ template, data, details });
+		} catch (error) {
+			await this.pdfFailed({ details, e: error as Error });
+		}
```

Comment:
```suggestion
		try {
			const room = await LivechatRooms.findOneById(details.rid);
			if (!room) {
				throw new Error('room-not-found');
			}
			const messages = await this.getMessagesFromRoom({ rid: room._id });

			const data = {
				visitor:
					room.v && (await LivechatVisitors.findOneById(room.v._id, { projection: { _id: 1, name: 1, username: 1, visitorEmails: 1 } })),
				agent: room.servedBy && (await Users.findOneAgentById(room.servedBy._id, { projection: { _id: 1, name: 1, username: 1 } })),
				closedAt: room.closedAt,
				messages: await this.getFiles(details.userId, messages),
			};

			await this.doRender({ template, data, details });
		} catch (error) {
			await this.pdfFailed({ details, e: error as Error });
		} finally {
			this.currentJobNumber--;
		}
```

There could be a chance that one of the databases calls before the try-catch and after `this.currentJobNumber++;` might fail, however in that case, the `currentJobNumber` won't get decremented because it is outside of try/catch loop. 
So maybe we can make use of try/catch/finally block here like this :top:

Reply:
:notbad: let me do the change